packages/abi-coder/src/interface.ts

```diff
@@ -20,27 +20,27 @@ export class Interface<TAbi extends JsonAbi = JsonAbi> {
   TODO: Refactor so that there's no need for externalLoggedTypes
    
   This is dedicated to external contracts added via `<base-invocation-scope.ts>.addContracts()` method. 
-  This is used to decode logs from contracts other than the main contract
+  This is used to decode logs from contracts ot her than the main contract
   we're interacting with.
   */
   private externalLoggedTypes: Record<string, Interface>;
-  jsonAbi: TAbi;
+  readonly jsonAbi: TAbi;
 
   constructor(jsonAbi: TAbi) {
     this.jsonAbi = jsonAbi;
 
     this.externalLoggedTypes = {};
 
     this.functions = Object.fromEntries(
-      jsonAbi.functions.map((x) => [x.name, new FunctionFragment(jsonAbi, x.name)])
+      jsonAbi.functions.map((x) => [x.name, new FunctionFragment(this.jsonAbi, x.name)])
```

Comment:
Let's try using either one of `jsonAbi` or `this.jsonAbi` consistently everywhere for more clarity.